bigquery/snippets/snippet_test.go

```diff
@@ -141,25 +155,8 @@ func TestAll(t *testing.T) {
 		t.Errorf("deleteTable(dataset:%q table:%q): %v", datasetID, dstTableID, err)
 	}
 
-	deleteDataset(t, ctx, datasetID)
-}
-
-func deleteDataset(t *testing.T, ctx context.Context, datasetID string) {
-	tc := testutil.SystemTest(t)
-	hc, err := google.DefaultClient(ctx, rawbq.CloudPlatformScope)
-	if err != nil {
-		t.Errorf("DefaultClient: %v", err)
-	}
-	s, err := rawbq.New(hc)
-	if err != nil {
-		t.Errorf("bigquery.New: %v", err)
-	}
-	call := s.Datasets.Delete(tc.ProjectID, datasetID)
-	call.DeleteContents(true)
-	call.Context(ctx)
-	if err := call.Do(); err != nil {
-		t.Errorf("deleteDataset(%q): %v", datasetID, err)
-	}
+	// cleanup dataset
+	client.Dataset(datasetID).DeleteWithContents(ctx)
```

Comment:
Is there an error to check from this?

Reply:
Moving this to a defer after dataset creation to make it clearer this isn't behavior under test, but test cleanup.